### PR TITLE
add docs on granting defineFunction access to defineData

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes:
Adds docs on configuring `defineFunction` access to the `defineData` schema. The experience of configuring the data client in the lambda is still a work in progress, but this adds documentation on the currently supported experience.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
Gen 2 function and data

Which platform(s) are affected by this PR (if applicable)?
NA

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx [hunk 1]
```diff
@@ -11,31 +12,143 @@ export function getStaticProps(context) {
...
+
+Function access can only be configured on the schema object. It cannot be configured on individual models or fields.
+
+</Callout>
+
+## Access the API using `aws-amplify`
+
+<Callout warning>
+
+**Experimental:** Configuring the `aws-amplify` data client within a Lambda function is under development. There are rough edges to the experience that we are actively working on.
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx:75]: Can we use a similar callout as the custom resource pages?
https://docs.amplify.aws/gen2/build-a-backend/add-aws-services/rest-api/
image

File: src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx [hunk 2]
```diff
@@ -11,31 +12,144 @@ export function getStaticProps(context) {
...
+// highlight-start
+// see instructions below on copying this file
+import { modelIntrospection } from './amplifyconfiguration.json';
+// highlight-end
+
+Amplify.configure(
+  {
+    API: {
+      GraphQL: {
+        endpoint: env.<amplifyData>_GRAPHQL_ENDPOINT!, // replace with your defineData name
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx:95]: is the non-null assertion needed?

Comment (edwardfoyle) [reply src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx:95]: ah good catch, that can be removed

File: src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx (hunk 1 above)
Comment (edwardfoyle) [reply src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx]: updated

**Review by josefaidt — DISMISSED** [review]

**Review by josefaidt — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**